### ISSUE TITLE
Fix bug that prevents GHA "Schedule Thursday 1100" from running

### ISSUE THREAD

**Issue by t-will-gillis** [hackforla/website#4768]
### Prerequisite

- You must be a member of Hack for LA. If you have not joined yet, please follow the steps on our [Getting Started](https://www.hackforla.org/getting-started) page.
- Please make sure you have read our **[Hack for LA Contributing Guide](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md)** before you claim/start working on an issue.  

### Overview
We need our GitHub actions to run on the schedule we set so that we can continue to maintain our workflow. For over a year the GHA "Schedule Thursday 1100" has failed before completion, and the source(s) of the error need to be addressed.  
  
  
### Details

> NOTE: the "Schedule Thursday 1100" triggers the GHA `trim-contributors` action and [contributors-data.js](github-actions/trigger-schedule/trim-contributors/contributors-data.js). **It appears that this file looks for inactive members and removes them from the "website" and/or "website-write" teams.**  

 
When reviewing HfLA "Actions" logs on the repo, it can be seen that the GHA "Schedule Thursday 1100" was been running weekly for almost a year. For the first Thursday of the month, the GHA fails. For the remaining Thursday, the GHA runs successfully because it is setting up the `trim-contributors` job, however that job is only scheduled to run on the first Thursday. In other words, these runs are unnecessary.  

Also, the paths on current Lines 38 and 45 are incorrect and need to be refactored. 

### Action Items

- [x] The [schedule-thu-1100.yml](https://github.com/t-will-gillis/website/edit/gh-pages/.github/workflows/schedule-thu-1100.yml) needs to be refactored so that the `trim-contributors` job is only being initiated on the first day of the month, and not every Thursday as it is currently. 
- [x] Delete the "Get current date" step from the .yml file. (There is a warning in the logs noting that the "set-output" command in the step "Get current date" is deprecated.)
- [x] _**Done**_ ~On the yml file, change versions: line 24 to `actions/checkout@v3` per PR 4735 and line 30 to `actions/setup-node@v3` per PR 4739~
- [x] The `working-directory` in the step "Install npm dependencies" needs to be refactored to the correct directory- i.e. `/github-data/`. (ref PR 3359).
- [x] The `run` path as shown in the step "Trim Members" needs to be refactored to the correct path to `contributors-data.js`. 
- [x] Lines 1 and 2 of [contibutors-data.js](https://github.com/hackforla/website/blob/gh-pages/github-actions/trigger-schedule/trim-contributors/contributors-data.js) are referring to modules in the `package.json` file and currently can't be found. To address this, either:
  - [x] Refactor the paths to `@octokit/rest` and `true-github-contributors` or if this isn't possible
  - [x] Move `contributors.js` to the `/github-data/` folder- and a.) remove the now empty `/trim-contributors/` folder and b.) adjust the path in the .yml accordingly.
- [x] Test the action on your own Project Board using `on: workflow_dispatch`. 
- [x] When this issue is merged, create new issue to add this GHA to the wiki. Please refer to [How to Contribute to the Wiki](https://github.com/hackforla/website/wiki/How-to-Contribute-to-the-Wiki) for instructions about where to place "Content to be Added/Edited in the New website-wiki Repo." 

Merge Team/ Leads:
- [x] _**Done**_ ~When this issue is merged, can immediately approve and merge #4739 (if testing fixes bug on the GHA with the new version number, this is evidence that changing the version number is acceptable)~
- [x]  _**Done**_ ~The `trim-contributors` job is the only one running in `schedule-thu-1100.yml`, and it is running only once a month. Should the job move to `schedule-monthly.yml`?~ Answer is yes, this job is moved to `schedule-monthly.yml`

### Resources/Instructions

- [schedule-thu-1100.yml](https://github.com/t-will-gillis/website/edit/gh-pages/.github/workflows/schedule-thu-1100.yml)

### Additional Resources
- Never done GitHub actions? [Start here!](https://docs.github.com/en/actions)
- [Events that trigger workflows](https://docs.github.com/en/actions/reference/events-that-trigger-workflows)
- [Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions)
- [actions/github-script](https://github.com/actions/github-script)

**Comment by ronaldpaek** [comment]
1. Jun 23-25
2. Jun 26

**Comment by ronaldpaek** [comment]
@t-will-gillis if the trim contributors action is moved to / merged to monthly, do we also need to ensure that the trim is happening only the 7th only? or can it be on the first?

**Comment by t-will-gillis** [comment]
@ronaldpaek For now, we should have this continue to run on the first Thursday of the month. The current cron only completes the job on the first Thursday of the month, but still runs nearly to the end all other Thursdays then stops short. Ideally we should be able to schedule the cron to only run on the first Thursday - but let's discuss. 
```

**Comment by ronaldpaek** [comment]
@t-will-gillis should I un assign myself to this?

**Comment by ronaldpaek** [comment]
@t-will-gillis I unassigned myself to this issue, I think it's beyond my scope and you can verify the other details, thanks :)

**Comment by t-will-gillis** [comment]
After several work sessions with @ronaldpaek, some questions are coming up so he has unassigned from the issue and I am moving it back to "New Issue Approval". We need to determine the intent for having the "schedule-thu-1100" automation, and compare this with the "schedule-monthly" automation to determine if there is a reason to have both actions since both seem to do much of the same things (such as generating the "Review Inactive Members" issue)

**Comment by ExperimentsInHonesty** [comment]
# Text for message

$assignee Our team bot has determined that you have not been active in the last 30 days.  You have been unassigned from all other issues and we are going to go ahead and close this issue and remove you from the write team.  

### If this is a mistake or you would like return to the Hack for LA website team
1. please respond in a comment with one of the two following messages

#### Our bot made a mistake, let us know
```
I have been active!  See Issue # or PR # 
```

#### I want to come back to the team
```
Please add me back to the write team, I am ready to work on an issue now.
```

### If its been less than 20 days
2. reopen this issue
3. move issue to in progress
4. make sure you are assigned to issue

### If its been more than 20 days
2. please send us a slack message with a link to your comment.

**Comment by ronaldpaek** [comment]
I have been active, how is this not coming to the conclusion I have been inactive for 30 days 🥲 @t-will-gillis do we need to fix some GitHub actions?

**Comment by t-will-gillis** [comment]
@ronaldpaek the "Text for message" comment from Bonnie is only an example of a message that this GHA should send in the future, i.e. it not apply to you/ you are not on the inactive list.
